File: language-support/hs/bindings/examples/nim-console/src/Logging.hs
-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
-- SPDX-License-Identifier: Apache-2.0


module Logging (Mes,noMes,tagMes,colourMes,plainMes,colourWrap) where

import System.Console.ANSI(
    setSGRCode, Color(..), SGR(SetColor), ConsoleLayer(Foreground), ColorIntensity(Vivid),)

type Mes = String -> IO ()

noMes :: Mes
noMes _ = return ()

plainMes :: Mes
plainMes = putStrLn

colourWrap :: Color -> String -> String
colourWrap col s =
  setSGRCode [SetColor Foreground Vivid col] <> s <>
  setSGRCode [SetColor Foreground Vivid White]

colourMes :: Color -> Mes -> Mes
colourMes col mes s = mes (colourWrap col s)

tagMes :: String -> Mes -> Mes
tagMes tag mes s = mes (tag <> s)

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/services/grpc/Logging.scala
// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.platform.server.api.services.grpc

import com.daml.logging.entries.LoggingEntry

object Logging {

  def traceId(id: Option[String]): LoggingEntry =
    "tid" -> (id.getOrElse(""): String)

}
